apps/frontend/src/app/2_molecules/AmountRenderer/AmountRenderer.tsx

```diff
@@ -60,24 +61,29 @@ export const AmountRenderer: FC<AmountRendererProps> = ({
     });
   }, [addNotification, value]);
 
-  const formattedValue = useMemo(() => {
-    const numberValue = Number(value);
-    const decimals = getDecimalPartLength(numberValue);
+  const countUpValues = useMemo(() => {
+    let endValue = String(value);
 
-    if (decimals > precision) {
-      return Number(numberValue.toString().slice(0, -(decimals - precision))); // trimming the difference between the number of decimals we have and the precision we want, cannot use toFixed() as we need to round down
+    if (typeof value === 'number' && isScientificNumber(value)) {
+      endValue = value.toFixed(18);
     }
-    return numberValue;
+
+    const [whole = '', decimals = ''] = endValue.split('.');
+    const end = parseFloat(
+      (whole ?? 0) + '.' + (decimals ?? 0).slice(0, precision),
+    );
+
+    return {
+      end,
+      decimals: getDecimalPartLength(end),
+      thousandSeparator: thousand,
+      decimalSeparator: decimal,
+    };
   }, [precision, value]);
 
   const localeFormattedValue = useMemo(
-    () => formatValue(formattedValue, precision), // we need to format formattedValue to make sure we round down (formatValue function rounds up)
-    [formattedValue, precision],
-  );
-
-  const formattedValueDecimals = useMemo(
-    () => getDecimalPartLength(formattedValue),
-    [formattedValue],
+    () => formatValue(value, precision), // we need to format formattedValue to make sure we round down (formatValue function rounds up)
```

Comment:
```suggestion
    () => formatValue(value, precision),
```
The comment is not relevant anymore as you don't format `formattedValue` but it's true that we need to round down. I was looking at `formatValue` and I think there is a bug in it and we should set the default `roundingMode` to `floor`/`trunc` instead of `halfExpand`. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#rounding_modes for the reference.

Reply:
@creed-victor What about this comment, have you looked into it?

Reply:
@tiltom, changed it to use `trunc`